clang/test/CIR/IR/alloca.cir

```diff
@@ -0,0 +1,29 @@
+// Test the CIR operations can parse and print correctly (roundtrip)
+
+// RUN: cir-opt %s | cir-opt | FileCheck %s
+!s32i = !cir.int<s, 32>
+!u64i = !cir.int<u, 64>
+
+module  {
+  cir.func @foo(%arg0: !s32i) {
+    %0 = cir.alloca !s32i, cir.ptr <!s32i>, ["len", init] {alignment = 4 : i64}
+    cir.store %arg0, %0 : !s32i, cir.ptr <!s32i>
+    %1 = cir.load %0 : cir.ptr <!s32i>, !s32i
+    %2 = cir.cast(integral, %1 : !s32i), !u64i 
+    %3 = cir.alloca !s32i, cir.ptr <!s32i>, %2 : !u64i, ["tmp"] {alignment = 16 : i64}
+    cir.return 
+  }
```

Comment:
Simplify, if possible, these tests to just:

```mlir
cir.func @foo(%arg0: !u64i) {
  %0 = cir.alloca !s32i, cir.ptr <!s32i>, ["len", init] {alignment = 4 : i64}
  %1 = cir.alloca !s32i, cir.ptr <!s32i>, %arg0 : !u64i, ["tmp"] {alignment = 16 : i64}
  cir.return 
}
```

The rationale here is to minimize uses of operations other than `cir.alloca` in this test so that it does not break if some other operation (e.g. `cir.cast`) changes.

Reply:
In general I'd agree, but in CIRGen we usually keep allocas in the beginning of blocks. This new type of allocation can have even reduced scope, and it's probably interesting to testcase that they could show up anywhere inside a scope that are not at the beginning of the bb.

Reply:
Ok, sounds good.